controller/volume_controller.go

```diff
@@ -2878,12 +2878,19 @@ func (c *VolumeController) checkAndFinishVolumeRestore(v *longhorn.Volume, e *lo
 		return nil
 	}
 
-	allScheduledReplicasIncluded, err := c.checkAllScheduledReplicasIncluded(v, e, rs)
+	allScheduledReplicasIncluded, healthReplicaCount, err := c.checkAllScheduledReplicasIncluded(v, e, rs)
 	if err != nil {
 		return err
 	}
 
-	if !isPurging && (v.Status.Robustness == longhorn.VolumeRobustnessHealthy || v.Status.Robustness == longhorn.VolumeRobustnessDegraded) && allScheduledReplicasIncluded {
+	allowCreateDegraded, err := c.ds.GetSettingAsBool(types.SettingNameAllowVolumeCreationWithDegradedAvailability)
+	if err != nil {
+		return err
+	}
+
+	if !isPurging &&
+		(v.Status.Robustness == longhorn.VolumeRobustnessHealthy || v.Status.Robustness == longhorn.VolumeRobustnessDegraded) &&
+		(allScheduledReplicasIncluded || (allowCreateDegraded && healthReplicaCount > 0)) {
```

Comment:
I just noticed the original condition already having `v.Status.Robustness == longhorn.VolumeRobustnessDegraded`, so should we still check `allowCreateDegraded`? 

or just below, good enough?

```go
if !isPurging &&
		(v.Status.Robustness == longhorn.VolumeRobustnessHealthy || v.Status.Robustness == longhorn.VolumeRobustnessDegraded) {
```

Reply:
We still need to check `allScheduledReplicasIncluded` and `degradedVolumeSupported` if users disable the global setting `allow-volume-creation-with-degraded-availability`.